docs/research/kb/reports/agents/logging-stack-research.md

```
@@ -0,0 +1,401 @@
+        if raw is None: continue          # fall through to the Struct's own default
+        kwargs[f.name] = msgspec.convert(raw, f.type, strict=False, dec_hook=_dec_hook)
+    return msgspec.convert(kwargs, cls, strict=False, dec_hook=_dec_hook)
+```
+
+Run against a faithful port of the real settings module — every field correct
+(prefixes, nesting, defaults preserved, `str→int`, `str→bool`, `Path`).
+**Failure arm armed:** a malformed port value → `ValidationError`.
+
+⭐ **`pydantic-settings` ships 14 sources; the settings module uses ONE.**
+
+```
```

markdownlint-cli2 (0.23.2)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)